src/Store.cpp

```diff
@@ -9,97 +10,169 @@
+const char lowCoinsMessage[] = " Insufficient Coins !";
+const char invalidChoice[] = " Invalid Choice !";
 
 void Store::StoreFront(Player* _Player) {
+	// Displays store menu and allows player to buy/sell items.
 
-	unsigned short choice = 0;
-	unsigned short coins;
+	bool isValidChoice = false;
 
 	do
 	{
-		ClearScreen();
-		coins = _Player->GetCoins();
-		cout << endl << endl;
-		_Player->DisplayInventory();
-		cout << endl << " Jeremy's Tools Store" << endl;
-		cout << " \"" << random_greeting() << "\"" << endl << endl;
-		cout << " ----ITEM----------------COST------ " << endl << endl;
-		cout << " 1) Arrows x 5\t\t " << arrowsCost << endl;
-		cout << " 2) Bomb\t\t " << bombCost << endl;
-		cout << " 3) Potion\t\t " << potionCost << endl;
-		cout << " 4) Whetstone\t\t " << whetstoneCost << endl << endl;
-		cout << " 0) Exit" << endl << endl;
-		cout << " ---------------------------------- " << endl << endl;
-		cout << " What do you want to buy today?" << endl << endl << " ";
-		choice = input();
+		// Display the store's main menu
+		DisplayStoreMainMenu(_Player);
+
+		// Input choice
+		unsigned short choice = input();
 		cout << endl;
+
+		isValidChoice = true;
 		switch (choice)
 		{
-		case ITEMTYPE::ARROWS:
-			if (coins < arrowsCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " 5 Arrows purchased ! " << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::ARROWS);
-				_Player->LoseCoins(arrowsCost);
-			}
-			Sleep(SLEEP_MS);
+		case 1: // Buy
+			BuyFront(_Player);
 			break;
-		case ITEMTYPE::BOMB:
-			if (coins < bombCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Bomb purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::BOMB);
-				_Player->LoseCoins(bombCost);
-			}
-			Sleep(SLEEP_MS);
-			break;
-		case ITEMTYPE::POTION:
-			if (coins < potionCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Potion purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::POTION);
-				_Player->LoseCoins(potionCost);
-			}
-			Sleep(SLEEP_MS);
+		case 2: // Sell
+			SellFront(_Player);
 			break;
-		case ITEMTYPE::WHETSTONE:
-			if (coins < whetstoneCost)
-			{
-				cout << lowCoinsMessage << endl;
-			}
-			else
-			{
-				cout << " Whetstone purchased !" << endl;
-				_Player->AddStoreItemToInventory(ITEMTYPE::WHETSTONE);
-				_Player->LoseCoins(whetstoneCost);
-			}
-			Sleep(SLEEP_MS);
-			break;
-		case 0://Exit Store
+		case 0: // Exit
 			break;
 		default:
 			cout << invalidChoice << endl;
+			isValidChoice = false;
 			Sleep(SLEEP_MS);
+			break;
+		}
+	} while (!isValidChoice);
+}
+
+void Store::BuyFront(Player* _Player) {
+	// Displays the buy front
+
+	const static catalogue_t stock_items = {
+		{ITEMTYPE::WHETSTONE, {"Whetstone", 1, 20}},
+		{ITEMTYPE::POTION, {"Potion", 1, 150}},
+		{ITEMTYPE::BOMB, {"Bomb", 1, 100}},
+		{ITEMTYPE::ARROWS, {"Arrows", 5, 40}}
+	};
```

Comment:
Could you make this a private member variable, also _not_ const? Reason is, as the game progresses new items may be added to the store.